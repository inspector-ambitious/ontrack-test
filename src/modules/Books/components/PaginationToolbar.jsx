import React from "react";
import PropTypes from "prop-types";
import { Pagination } from "react-bootstrap";

const PaginationToolbar = ({ page, setPage, maxPage }) => {
  return (
    <Pagination>
      <Pagination.First disabled={page === 1} onClick={() => setPage(1)} />
      <Pagination.Prev disabled={page < 2} onClick={() => setPage(page - 1)} />
      <Pagination.Item active>
        Page: {page}/{maxPage}
      </Pagination.Item>
      <Pagination.Next
        disabled={page >= maxPage}
        onClick={() => setPage(page + 1)}
      />
      <Pagination.Last
        disabled={page === maxPage}
        onClick={() => setPage(maxPage)}
      />
    </Pagination>
  );
};

PaginationToolbar.propTypes = {
  page: PropTypes.number.isRequired,
  setPage: PropTypes.func.isRequired,
  maxPage: PropTypes.number.isRequired,
};

export default PaginationToolbar;
