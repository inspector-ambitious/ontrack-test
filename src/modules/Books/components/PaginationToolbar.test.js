import React from "react";
import { shallow } from "enzyme";
import PaginationToolbar from "./PaginationToolbar";

describe("Books/components/PaginationToolbar", () => {
  const setPage = jest.fn();
  const defaultProps = {
    page: 2,
    maxPage: 7,
    setPage,
  };
  beforeEach(() => {
    jest.resetAllMocks();
  });
  it("should render without crashing", () => {
    const wrapper = shallow(<PaginationToolbar {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  describe("First", () => {
    it("should be disabled if the current page is 1", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} page={1} />);
      expect(wrapper.find("First").props().disabled).toBe(true);
    });
    it("should call set the page to 1 on click", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} />);
      wrapper.find("First").simulate("click");
      expect(setPage).toHaveBeenCalledTimes(1);
      expect(setPage).toHaveBeenCalledWith(1);
    });
  });
  describe("Prev", () => {
    it("should be disabled if the current page is 1", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} page={1} />);
      expect(wrapper.find("Prev").props().disabled).toBe(true);
    });
    it("should set the page to (current page - 1) on click", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} />);
      wrapper.find("Prev").simulate("click");
      expect(setPage).toHaveBeenCalledTimes(1);
      expect(setPage).toHaveBeenCalledWith(1);
    });
  });
  describe("Last", () => {
    it("should be disabled if the current page is equal to maxPage", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} page={7} />);
      expect(wrapper.find("Last").props().disabled).toBe(true);
    });
    it("should set the page to max page on click", () => {
      const wrapper = shallow(<PaginationToolbar {...defaultProps} />);
      wrapper.find("Last").simulate("click");
      expect(setPage).toHaveBeenCalledTimes(1);
      expect(setPage).toHaveBeenCalledWith(7);
    });
  });
});
