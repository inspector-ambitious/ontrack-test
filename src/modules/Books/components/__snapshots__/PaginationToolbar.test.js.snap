// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Books/components/PaginationToolbar should render without crashing 1`] = `
<ForwardRef>
  <First
    disabled={false}
    onClick={[Function]}
  />
  <Prev
    disabled={false}
    onClick={[Function]}
  />
  <PageItem
    active={true}
    activeLabel="(current)"
    disabled={false}
  >
    Page: 
    2
    /
    7
  </PageItem>
  <Next
    disabled={false}
    onClick={[Function]}
  />
  <Last
    disabled={false}
    onClick={[Function]}
  />
</ForwardRef>
`;
